zapier/babel-preset-zapier
# Fix issues with transpiled code in Edge

This fixes two issues with Edge and the code that babel outputs:

1. Edge doesn't fully support default parameters but `preset-env` doesn't transpile them. We need to turn on `plugin-transform-parameters` manually to fix this
2. Edge has a bug with ES6 classes where a `super()` call from a class with a non-ES6 class superclass doesn't return `this` properly (See [this comment](https://github.com/zapier/zapier/pull/21319#issuecomment-437871079) for more info). We need to turn on `plugin-transform-classes` to fix this.

2 is quite a bummer since not transpiling classes was probably a big save in terms of kb's. Hopefully this is something we can get rid of soon :(

## Testing

1. To test, use the reviewlab from this PR: https://github.com/zapier/zapier/pull/20092 (https://zapier-lab-20092.k8s.zapier.com/)
1. Use browserstack and test Edge 17 and 16
1. Make sure you have the caching disabled on Edge: 
![](https://cdn.zapier.com/storage/photos/ac7708c5d15f061d5affada8c504fd3b.png)
1. Make sure you also check that loading the editor is working correctly
